### PR TITLE
Feature: support shadow type match regex (#365)

### PR DESCRIPTION
**Pull request by csynineyang**
1. create shadow tables in init.sql & sharding.sql
2. hide shadow tables for "show tables" command
3. impl shadow match hint rule (only select)
4. support shadow type match regex (only select)

Test Cases:
SELECT name FROM student WHERE uid = 1; //sharding
SELECT name FROM student WHERE name = 'lilei'; //fullscan
SELECT name FROM student WHERE uid = 1 and name = 'lilei'; //sharding
SELECT name FROM student WHERE uid = 1 or name = 'lilei'; //fullscan

/*A! shadow(shadow) */ SELECT name FROM student WHERE uid = 1; //sharding shadow tables
/*A! shadow(shadow) */ SELECT name FROM student WHERE name = 'lilei'; //fullscan shadow tables
/*A! shadow(shadow) */ SELECT name FROM student WHERE uid = 1 and name = 'lilei'; //sharding shadow tables
/*A! shadow(shadow) */ SELECT name FROM student WHERE uid = 1 or name = 'lilei'; //fullscan shadow tables

regex：only support eq & ne
SELECT name FROM student WHERE name = 'hanmeimei'; //fullscan shadow tables
SELECT name FROM student WHERE uid = 1 and name = 'hanmeimei'; //sharding shadow tables
SELECT name FROM student WHERE uid = 1 or name = 'hanmeimei'; //fullscan shadow tables
SELECT name FROM student WHERE NOT name = 'hanmeimei'; //fullscan

**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/runtime/ast/expression_atom.go [hunk 1]
```diff
@@ -132,7 +132,7 @@ func (u *UnaryExpressionAtom) Accept(visitor Visitor) (interface{}, error) {
 }
 
 func (u *UnaryExpressionAtom) IsOperatorNot() bool {
-	switch u.Operator {
+	switch strings.ToUpper(strings.Trim(u.Operator, " ")) {
```
Comment (jjeffcaii) [inline pkg/runtime/ast/expression_atom.go:135]: Use `TrimSpace` instead.

**Review by jjeffcaii — APPROVED** [review]
pls add ut

**Review by dongzl — APPROVED** [review]

**Review by Lvnszn — APPROVED** [review]
looks like need add create table script

**Comment by csynineyang** [comment]
> looks like need add create table script

see sharding.sql

File: pkg/runtime/ast/expression_atom.go (hunk 1 above)
Comment (csynineyang) [reply pkg/runtime/ast/expression_atom.go]: done

**Comment by csynineyang** [comment]
> pls add ut

add integration test for shadow tables

**Pull request merged**